src/libstore/filetransfer.cc

```diff
@@ -101,6 +101,7 @@ struct curlFileTransfer : public FileTransfer
                     this->result.data.append(data);
               })
         {
+            requestHeaders = curl_slist_append(requestHeaders, "Accept-Encoding: zstd, br, gzip, deflate");
```

Comment:
We can't easily get this list from curl or at compile time?

Reply:
@SuperSandro2000 this is actually not curl decompressing but nix has its own decompression sinks below. In fact my first version was just doing `curl_easy_setopt(curl, CURLOPT_ACCEPT_ENCODING, "");` and disable having nix doing the decompression afterwards (or else it would break).

Reply:
Except from `brotli`, the decompression is transparently handled by `libarchive` IIRC. So ideally we could just query `libarchive` for that, but after a quick search I couldn't find a way to do that.

Reply:
@thufschmitt the alternative would be to let curl handle all of the decompression. Just make a call which way do you prefer.

Reply:
Dunno, I have hardly ever touched these parts of the codebase, and I've no idea why things are this way :man_shrugging: I guess @edolstra would have a more informed opinion here.

What would be the cost of letting `curl` handle it? Should it just be a matter of setting `CURLOPT_ACCEPT_ENCODING` or is there something more involved?

Reply:
From what I remember, curl doesn't support all the formats that Nix supports (e.g. bzip2, and I think the curl in nixpkgs didn't have brotli enabled at the time). And Nix has to have its own support for these formats anyway. So there isn't much advantage to letting curl handle the decompression.

Reply:
> I think the curl in nixpkgs didn't have brotli enabled at the time

I changed that a good while ago and moved brotili support from curlFull to curl.

So we can just keep this as is and maybe need to update the values manual in the future due to the lack of better options?